### PR TITLE
Regex config options

### PR DESCRIPTION
**Pull request by toondaey**

### CODE REVIEW

File: cognite/extractorutils/configtools/elements.py [hunk 1]
```diff
@@ -696,3 +696,39 @@ def create_state_store(
             return LocalStateStore(file_path="states.json", cancellation_token=cancellation_token)
         else:
             return NoStateStore()
+
+
+class RegExpFlag(Enum):
+    IGNORECASE = "ignore-case"
+    IC = "i"
+    ASCII = "ascii-only"
+    A = "a"
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:705]: Why two of each?

Comment (toondaey) [reply cognite/extractorutils/configtools/elements.py:705]: For backward compatibility in the file extractor.

Comment (mathialo) [reply cognite/extractorutils/configtools/elements.py:705]: Ah, I see. That's quite annoying. If we want to preserve backwards compatibility we'd have to change the `options` back to `flags` also I guess.

I don't think our current dataclass/dacite stack supports aliases, which is what we should have used here.

Comment (toondaey) [reply cognite/extractorutils/configtools/elements.py:705]: We could support both `flags` and `options` at the same time but indicate which one will be deprecated in the next release cycle, maybe...

**Review by mathialo — APPROVED** [review]

**Pull request merged**